api/test/trace/propagation/http_text_format_test.cc

```diff
@@ -0,0 +1,190 @@
+#include "opentelemetry/trace/propagation/http_text_format.h"
+#include "opentelemetry/trace/propagation/http_trace_context.h"
+#include "opentelemetry/context/context.h"
+#include "opentelemetry/trace/span.h"
+#include "opentelemetry/trace/default_span.h"
+#include "opentelemetry/trace/span_context.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/nostd/span.h"
+#include "opentelemetry/nostd/shared_ptr.h"
+#include "opentelemetry/trace/trace_id.h"
+
+
+#include <map>
+#include <memory>
+#include <string>
+
+#include <gtest/gtest.h>
+
+using namespace opentelemetry;
+
+static nostd::string_view Getter(const std::map<std::string,std::string> &carrier, nostd::string_view trace_type = "traceparent") {
+    auto it = carrier.find(std::string(trace_type));
+    if (it != carrier.end()) {
+        return nostd::string_view(it->second);
+    }
+    return "";
+}
+
+static void Setter(std::map<std::string,std::string> &carrier, nostd::string_view trace_type = "traceparent", nostd::string_view trace_description = "") {
+    carrier[std::string(trace_type)] = std::string(trace_description);
+}
+
+static trace::propagation::HttpTraceContext<std::map<std::string,std::string>> format = trace::propagation::HttpTraceContext<std::map<std::string,std::string>>();
+
+static nostd::string_view trace_id = "12345678901234567890123456789012";
+static nostd::string_view span_id = "1234567890123456";
+
+using map_http_trace_context = trace::propagation::HttpTraceContext<std::map<std::string,std::string>>;
+TEST(HTTPTextFormatTest, TraceIdBufferGeneration)
+{
+    constexpr uint8_t buf[] = {1, 2, 3, 4, 5, 6, 7, 8, 8, 7, 0xaa, 0xbb, 0xcc, 0xdd, 0xee, 0xff};
+    trace::TraceId id(buf);
+    EXPECT_EQ(map_http_trace_context::GenerateTraceIdFromString("01020304050607080807aabbccddeeff"),id);
+}
+
+TEST(HTTPTextFormatTest, SpanIdBufferGeneration)
+{
+    constexpr uint8_t buf[] = {1, 2, 0xaa, 0xbb, 0xcc, 0xdd, 0xee, 0xff};
+    trace::SpanId id(buf);
+    EXPECT_EQ(map_http_trace_context::GenerateSpanIdFromString("0102aabbccddeeff"),id);
+}
+
+TEST(HTTPTextFormatTest, TraceFlagsBufferGeneration)
+{
+    trace::TraceFlags flags;
+    EXPECT_EQ(map_http_trace_context::GenerateTraceFlagsFromString("00"),flags);
+}
+
+TEST(HTTPTextFormatTest, HeadersWithTraceState)
+{
+    const std::map<std::string,std::string> carrier = {{"traceparent","00-4bf92f3577b34da6a3ce929d0e0e4736-0102030405060708-01"},{"tracestate","congo=congosSecondPosition,rojo=rojosFirstPosition"}};
+    nostd::shared_ptr<trace::Span> sp{new trace::DefaultSpan()};
+    context::Context ctx1 = context::Context("current-span",sp);
+    context::Context ctx2 = format.Extract(Getter,carrier,ctx1);
+    std::map<std::string,std::string> c2 = {};
+    format.Inject(Setter,c2,ctx2);
+    EXPECT_EQ(std::string(c2["traceparent"]),"00-4bf92f3577b34da6a3ce929d0e0e4736-0102030405060708-01");
```

Comment:
Does the EXPECT_EQ work on nostd::string_view (i.e. without constructing a string)?